src/main/java/org/jenkinsci/plugins/workflow/steps/RetryStepExecution.java

```diff
@@ -64,6 +65,11 @@ public void onFailure(StepContext context, Throwable t) {
                     TaskListener l = context.get(TaskListener.class);
                     if (t instanceof AbortException) {
                         l.error(t.getMessage());
+                    } else if (t instanceof FlowInterruptedException) {
+                        FlowInterruptedException fie = (FlowInterruptedException) t;
+                        for (CauseOfInterruption cause : fie.getCauses()) {
+                            l.getLogger().println(cause.getShortDescription());
```

Comment:
```suggestion
                            cause.print(l);
```

is better as the impl can include hyperlinks and the like.